compiler/rustc_hir/src/hir.rs

```diff
@@ -289,6 +290,11 @@ pub enum ConstArgKind<'hir> {
     /// However, in the future, we'll be using it for all of those.
     Path(QPath<'hir>),
     Anon(&'hir AnonConst),
+    /// **Note:** Not all const inference variables are represented as
+    /// `ConstArgKind::Infer`. In cases where it is ambiguous whether
+    /// a generic arg is a type or a const, inference variables are
+    /// represented as `GenericArg::Infer` instead.
+    Infer(InferArg),
```

Comment:
Could you add a similar comment to `TyKind::Infer` and `GenericArg::Infer` as those are also places where people may get confused and think it's the only way to represent `Infer` when it's not :3